setup.py

```diff
@@ -56,14 +56,24 @@ def detect_distutils():
         else:
             amb_scripts.append('scripts/ambuild_objcopy_wrapper.sh')
 
-    setup(name = 'AMBuild',
-          version = '2.0',
-          description = 'AlliedModders Build System',
-          author = 'David Anderson',
-          author_email = 'dvander@alliedmods.net',
-          url = 'http://www.alliedmods.net/ambuild',
-          packages = find_packages(),
-          python_requires = '>=2.6',
-          entry_points = {'console_scripts': ['ambuild = ambuild2.run:cli_run']},
-          scripts = amb_scripts,
-          zip_safe = False)
+    setup(
+        name = 'AMBuild',
+        description = 'AlliedModders Build System',
+        author = 'David Anderson',
+        author_email = 'dvander@alliedmods.net',
+        url = 'http://www.alliedmods.net/ambuild',
+        packages = find_packages(),
+        python_requires = '>=2.7',
+        dependency_links = [
+            'git+https://github.com/WildCard65/setuptools-git-versioning.git#egg=setuptools-git-versioning-1.1.7.1'

```

Comment:
If a user wants to install another package that depends on a later version of setuptools-git-versioning... won't this break?

Having a dependency on a user repo is a bit concerning... from a security/maintenance perspective, that could be anything, or change to anything, or get removed, at any time. So if we have to do this, it'd be better to pull it into an official AM repository. (But then we pretty much have to carry it around forever, which is also annoying.)

Reply:
It's a stop-gap for now (I'll submit a PR to dolfinus/setuptools-git-versioning with the changes, but I can drop the "<1.1.8" requirement if you want, but this'll break 3.3 again because of the "typing" requirement (as pip will pull from pypi for newer versions)

Reply:
Unless there's some super pressing need, I think it's best to wait and make sure it's done correctly.

If Python 2.7 works, we can talk about minimum Python 3 versions. I think systems that tend to have older versions of Python 3 will definitely have Python 2.7, so maybe raising the minimum 3.x version is fine.

Reply:
Oh also, once the package is installed, "setup_requires" stuff is ignored.

Reply:
Here's the python version requirements of the typing backport: Python >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*
Python 3.3 is considered deprecated and EOL just like Python 2, just been EOL longer than Python 2